scylladb/scylla-manager
# Autogenerate sctool docs from yaml files

Closes https://github.com/scylladb/scylla-manager/issues/2930

Autogenerates ``sctool`` docs from YAML files using the extension ``sphinxcontrib.datatemplates``.

## How to test this PR

1. Clone this PR. For more information, see [Cloning pull requests locally](https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/checking-out-pull-requests-locally).

2. Enter the docs folder, and run:

```
make preview
````

3. Open http://127.0.0.1:5500/sctool/ with your favorite browser and preview the docs.


## Before merging this PR ...

Would it be possible to update the autogenerated YAML files? 

1. Sphinx needs a special character to detect lists. For example:

```
    '*' - matches any number of any characters including none
    '?' - matches any single character
    '[abc]' - matches one character given in the bracket
    '[a-z]' - matches one character from the range given in the bracket
```

Must be:

```
    * '*' - matches any number of any characters including none
    * '?' - matches any single character
    * '[abc]' - matches one character given in the bracket
    * '[a-z]' - matches one character from the range given in the bracket
```

2. To highlight inline-code (e.g. ``abc``), replace the single quotes as follows ``'abc'`` -> ` ``[abc]`` `.

## Future improvements

* [Add a collapsible component](https://github.com/scylladb/sphinx-scylladb-theme/issues/276) in the theme to hide "inherited options".
* Support for more than one example per command. For example, instead of:

```
example: |
  sctool status -c prod-cluster
  Datacenter: eu-west
```

The example key could look like this:

```
examples: 
   - name: Example A 
      description: Example description
     command: sctool xyz
   - name: Example B
      description: Example description
     command: sctool xyz
```

Adding the new keys ``name`` and ``description`` will also be possible to format commands as code-blocks and descriptions as regular text.
